src/library/scala/collection/MapView.scala

```diff
@@ -55,6 +83,12 @@ object MapView {
     override def isEmpty: Boolean = underlying.isEmpty
   }
 
+  @SerialVersionUID(3L)
+  class Concat[K, +V](left: SomeMapOps[K, V], right: SomeMapOps[K, V]) extends AbstractMapView[K, V] {
+    def get(key: K): Option[V] = right.get(key).orElse(left.get(key))
+    def iterator: Iterator[(K, V)] = right.iterator.concat(left.iterator.filter { case (k, _) => !right.contains(k)})
```

Comment:
This is a slightly unexpected order for the concat of the maps.  Can you flip it around so the left map comes first (even though it's filtered because the later values take precedence)?

Reply:
Done